tests/test_config.py

```diff
@@ -0,0 +1,100 @@
+import pytest
+from pathlib import Path
+from staging_service.config import MissingAuthToken, StagingServiceConfig
+import os
+
+# separate these for easier assertions
+AUTH_URL = "https://example.com/auth"
+DATA_DIR = "/kb/deployment/data/bulk"
+META_DIR = "/kb/deployment/data/metadata"
+CONCIERGE_PATH = "/kbaseconcierge"
+FILE_EXTENSION_MAPPINGS = "/kb/deployment/file_mappings.json"
+DTS_MANIFEST_SCHEMA = "/kb/deployment/dts_manifest_schema.json"
+
+VALID_CONFIG = f"""
+[staging_service]
+AUTH_URL = {AUTH_URL}
+DATA_DIR = {DATA_DIR}
+META_DIR = {META_DIR}
+CONCIERGE_PATH = {CONCIERGE_PATH}
+FILE_EXTENSION_MAPPINGS = {FILE_EXTENSION_MAPPINGS}
+DTS_MANIFEST_SCHEMA = {DTS_MANIFEST_SCHEMA}
+"""
+
+
+def write_config_file(config_dir: Path, content: str) -> str:
+    """
+    Writes out some content to "config.cfg" in the provided
+    directory. Returns the path to the file as a string.
+    """
+    file = config_dir / "config.cfg"
+    file.write_text(content)
+    return str(file)
+
+
+def test_valid_config(tmp_path):
+    config_path = write_config_file(tmp_path, VALID_CONFIG)
+    config = StagingServiceConfig(config_path)
+
+    assert config.auth_url == AUTH_URL
+    assert config.data_dir == DATA_DIR
+    assert config.meta_dir == META_DIR
+    assert config.concierge_path == CONCIERGE_PATH
+    assert config.file_extension_mappings == FILE_EXTENSION_MAPPINGS
+    assert config.dts_manifest_schema == DTS_MANIFEST_SCHEMA
+    assert config.auth_token == os.environ["AUTH_TOKEN"]
+
+
+def test_missing_config_path():
+    with pytest.raises(ValueError, match="config_path is required"):
+        StagingServiceConfig("")
+
+
+def test_missing_config_file():
+    with pytest.raises(FileNotFoundError):
+        StagingServiceConfig("missing.cfg")
+
+
+def test_missing_heading(tmp_path):
+    bad_config = "[wrong_section]\nfoo=bar"
+    config_path = write_config_file(tmp_path, bad_config)
+    with pytest.raises(ValueError, match="missing required section"):
```

Comment:
```suggestion
    with pytest.raises(ValueError, match=f"{config_path} missing required section staging_service"):
```

making sure the user gets the info they need to debug

Reply:
Done.

Reply:
Personally I would make staging_service a literal in the string but that's fine

Reply:
:+1: